### PR TITLE
Add FeeSettings, LedgerHashes ledger object types

### PR DESCRIPTION
**Pull request by mDuo13**
Adds two more types of ledger objects that weren't documented until now. Both of them are sort of internal tracking systems, but they are useful in their own ways.

This also corrects the LedgerEntryType and space key values in a couple places. Even though both types are enumerated using single ASCII characters in the code base, they get cast to UInt16's when serialized, so I represented them as `0x00__` values to make it clearer. (This explains the hashes of the singletons more accurately.)

Also, this diagram didn't make it into the doc, but it explains the discrepancy in the LedgerEntryType and Space Key values for the LedgerHashes and FeeSettings types:
ledger-node-space-key-comparison

### CODE REVIEW

File: content/reference-ledger-format.md [hunk 1]
```diff
@@ -358,6 +362,118 @@ The ID of an Escrow object is the SHA-512Half of the following values put togeth
...
+   "ReserveIncrement": 5000000,
+   "index": "4BC50C9B0D8515D3EAAE1E74B29A95804346C491EE1A95BF25E4AAB854A6A651"
+}
+```
+
+The `FeeSettings` object has the following fields:
+
+| Name                | JSON Type | [Internal Type][] | Description            |
+|:--------------------|:----------|:------------------|:-----------------------|
+| `LedgerEntryType`   | String    | UInt16            | The value `0x0073`, mapped to the string `FeeSettings`, indicates that this object is a list of ledger hashes. |
```
Comment (JoelKatz) [inline content/reference-ledger-format.md]: The text here is not correct. Should read "indicates that this object contains the network fee settings" or something like that.

File: content/reference-ledger-format.md [hunk 2]
```diff
@@ -358,6 +362,118 @@ The ID of an Escrow object is the SHA-512Half of the following values put togeth
...
+| `ReserveBase`       | Number    | UInt32            | The [base reserve](concept-reserves.html#base-reserve-and-owner-reserve) for an account in the XRP Ledger, as drops of XRP. |
+| `ReserveIncrement`  | Number    | UInt32            | The incremental [owner reserve](concept-reserves.html#base-reserve-and-owner-reserve) for owning objects, as drops of XRP. |
+| `Flags`             | Number    | UInt32            | A bit-map of boolean flags for this object. No flags are defined for this type. |
+
+**Warning:** The JSON format for this ledger object type is unusual. The `BaseFee`, `ReserveBase`, and `ReserveIncrement` indicate drops of XRP but ***not*** in the usual format for [specifying XRP](reference-rippled.html#specifying-currency-amounts).
+
+### FeeSettings ID Format
+
+The `FeeSettings` object ID is the hash of the `FeeSettings` space key (`0x0065`) only. This means that the ID of the Amendments object in a ledger is always:
+
```
Comment (JoelKatz) [inline content/reference-ledger-format.md]: "Amendments" should be "FeeSettings"

**Review by JoelKatz — APPROVED** [review]
:+1: (With correcting the two mistakes I noted.)

File: content/reference-ledger-format.md [hunk 3]
```diff
@@ -358,6 +362,118 @@ The ID of an Escrow object is the SHA-512Half of the following values put togeth
...
+**Note:** The oldest "previous history" `LedgerHashes` object contains only 255 entries because the genesis ledger has ledger index 1, not 0.
+
+The "previous history" `LedgerHashes` objects act as a [skip list](https://en.wikipedia.org/wiki/Skip_list) so you can get the hash of any historical flag ledger from its index. From there, you can use that flag ledger's "recent history" object to get the hash of any other ledger.
+
+### LedgerHashes ID Formats
+[[Source]<br>](https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/Indexes.cpp#L28-L43)
+
+There are two formats for `LedgerHashes` object IDs, depending on whether the object is a "recent history" sub-type or a "previous history" sub-type.
+
+The **"recent history"** `LedgerHashes` object has an ID that is the SHA-512Half of the `LedgerHashes` space key (`0x0073`). In other words, the "recent history" always has the ID `B4979A36CDC7F3D3D5C31A4EAE2AC7D7209DDA877588B9AFC66799692AB0D66B`.
```
Comment (jbheron) [inline content/reference-ledger-format.md]: I hadn't heard of SHA-512Half, and had to look it up. Seeing it is a Ripple-specific term, I'd suggest adding a link to it.

**Review by jbheron — APPROVED** [review]
LGTM 

I added one suggestion about a link to SHA-512Half

**Pull request merged**